app/code/Magento/WishlistGraphQl/Model/Resolver/AddItemToWishlist.php

```diff
@@ -0,0 +1,130 @@
+     */
+    private $productCollectionFactory;
+
+    /**
+     * @param WishlistDataProvider $wishlistDataProvider
+     * @param SearchCriteriaInterface $searchCriteria
+     * @param Visibility $productVisibility
+     * @param StockProcessor $stockProcessor
+     * @param CollectionFactory $collectionFactory
+     */
+    public function __construct(
+        WishlistDataProvider $wishlistDataProvider,
+        SearchCriteriaInterface $searchCriteria,
+        Visibility $productVisibility,
+        StockProcessor $stockProcessor,
+        CollectionFactory $collectionFactory
+    ) {
+        $this->wishlistDataProvider = $wishlistDataProvider;
+        $this->searchCriteria = $searchCriteria;
+        $this->catalogProductVisibility = $productVisibility;
+        $this->stockProcessor = $stockProcessor;
+        $this->productCollectionFactory = $collectionFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (!isset($args['input']['skus'])) {
+            throw new GraphQlInputException(__('You must specify at least one "sku" value'));
+        }
+        $wishlist = $this->wishlistDataProvider->getWishlistForCustomer($context->getUserId());
+        if (!$wishlist || !$wishlist->getId()) {
+            throw new GraphQlInputException(__('Cannot get a wish list for the specified Customer ID'));
+        }
+        $this->addMultipleProducts($args['input']['skus'], $wishlist);
+        return [
+            'wishlist' => [
+                'sharing_code' => $wishlist->getSharingCode(),
+                'updated_at' => $wishlist->getUpdatedAt(),
+                'items_count' => $wishlist->getItemsCount(),
+                'name' => $wishlist->getName(),
+                'model' => $wishlist,
+            ]
+        ];
+    }
+
+    /**
+     * @param $skus
+     * @param Wishlist $wishList
+     *
+     * @return Wishlist
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function addMultipleProducts($skus, Wishlist $wishList)
+    {
+        $productCollection = $this->getProductCollectionBySkus($skus);
+        if ($productCollection->getSize()) {
+            foreach ($productCollection as $product) {
+                /** @var Product $product */
+                $wishList->addNewItem($product);
+            }
+        }
+        return $wishList;
+    }
+
+    /**
+     * @param array $skus
+     * @return Collection
+     */
+    private function getProductCollectionBySkus($skus)
+    {
+        /** @var Collection $productsCollection */
+        $productsCollection = $this->productCollectionFactory->create();
+        $productsCollection->setVisibility($this->catalogProductVisibility->getVisibleInSiteIds());
+        $productsCollection->addAttributeToFilter('sku', ['in' => $skus]);
+        return $productsCollection;
```

Comment:
THank you. That helps. Yes. I used this Helper `Magento\CatalogInventory\Helper\Stock` to add isInStockFilter to the collection. But it is deprecated. Will you @naydav suggest something else, since I haven't found an appropriate filter.